server/mscalendar/availability.go

```diff
@@ -276,19 +285,40 @@ func (m *mscalendar) setStatusFromCalendarView(user *store.User, currentStatus s
 	return message, nil
 }
 
-func (m *mscalendar) setStatusOrAskUser(user *store.User, status string, events []*remote.Event) error {
-	if user.Settings.GetConfirmation {
-		url := fmt.Sprintf("%s%s%s", m.Config.PluginURLPath, config.PathPostAction, config.PathConfirmStatusChange)
-		_, err := m.Poster.DMWithAttachments(user.MattermostUserID, views.RenderStatusChangeNotificationView(events, status, url))
-		if err != nil {
-			return err
+func (m *mscalendar) setStatusOrAskUser(user *store.User, status *model.Status, events []*remote.Event, isFree bool) error {
```

Comment:
```suggestion
func (m *mscalendar) setStatusOrAskUser(user *store.User, currentStatus *model.Status, events []*remote.Event, isFree bool) error {
```

The parameters to this function seem sort of cryptic. The fact that we need to access `user` to determine their previous status, and that it is a fundamental piece of the logic here, is a bit confusing.

For example, I would expect the `status` parameter to be the status we are setting the user to, but that parameter is actually the current status. We already know what status we need to set the user to in the previous function right? I suppose it's fine that this function is performing the logic, but it's a bit hard to read. Maybe put some comments in this function?